js/misc/paygManager.js

```diff
@@ -88,6 +112,57 @@ function _isPaygEnabled() {
     return false;
 }
 
+// Takes an UNIX timestamp (in seconds) and returns a string
+// with a precision level appropriate to show to the user.
+//
+// The returned string will be formatted just in seconds for times
+// under 1 minute, in minutes for times under 2 hours, in hours and
+// minutes (if applicable) for times under 1 day, and then in days
+// and hours (if applicable) for anything longer than that in days.
+//
+// Some examples:
+//   - 45 seconds => "45 seconds"
+//   - 60 seconds => "1 minute"
+//   - 95 seconds => "1 minute"
+//   - 120 seconds => "2 minutes"
```

Comment:
For this case, I would probably follow the same rule that you use for showing number of minutes under 2 hours, and show the full number of seconds if under 2 minutes.  But I don't feel that strongly, so I'm going to merge as is.